.github/workflows/ci-gh1.yml

```
@@ -0,0 +1,60 @@
+name: Build legate.core on GH
+
+concurrency:
+  group: ci-gpu-on-${{ github.event_name }}-from-${{ github.ref_name }}
+  cancel-in-progress: true
+
+on:
+  push:
+    branches-ignore:
+      - gh-pages  # deployment target branch (this workflow should not exist on that branch anyway)
+  # NOTE: this workflow will not work for pull_requests since GH pull requests do not have access to secrets.
+
+jobs:
+  build:
+    permissions: read-all
+    # Ref: https://docs.rapids.ai/resources/github-actions/#cpu-labels for `linux-amd64-cpu4`
+    runs-on: ${{ github.repository == 'nv-legate/legate.core' && 'linux-amd64-cpu4' || 'ubuntu-latest' }}
+    container:
+      options: -u root
+      image: rapidsai/devcontainers:23.06-cpp-cuda11.8-mambaforge-ubuntu22.04
+      env:
+        PYTHONDONTWRITEBYTECODE: 1
+        SCCACHE_REGION: us-east-2
+        SCCACHE_BUCKET: rapids-sccache-devs
+        SCCACHE_S3_KEY_PREFIX: legate-cunumeric-dev
+        VAULT_HOST: https://vault.ops.k8s.rapids.ai
```

If we want to make it work without secrets, we could do:
```suggestion
        VAULT_HOST: ${{ secrets.PERSONAL_ACCESS_TOKEN && 'https://vault.ops.k8s.rapids.ai' || '' }
```